fiftyone/core/collections.py

```diff
@@ -10622,15 +10622,22 @@ def _is_full_collection(self):
         if isinstance(self, fod.Dataset) and self.media_type != fom.GROUP:
             return True
 
-        # pylint:disable=no-member
-        if (
-            isinstance(self, fov.DatasetView)
-            and self._dataset.media_type == fom.GROUP
-            and len(self._stages) == 1
-            and isinstance(self._stages[0], fos.SelectGroupSlices)
-            and self._pipeline() == []
-        ):
-            return True
+        if isinstance(self, fov.DatasetView):
+            # pylint:disable=no-member
+            if (
+                not len(self._stages)
+                and not self._is_clips
+                and self._dataset.media_type != fom.GROUP
+            ):
+                return True
+
+            if (
+                self._dataset.media_type == fom.GROUP
+                and len(self._stages) == 1
+                and isinstance(self._stages[0], fos.SelectGroupSlices)
+                and not self._pipeline()
+            ):
+                return True
```

Comment:
lets add comments about the cases
```suggestion
        if isinstance(self, fov.DatasetView):
            # Case 2: Empty non-group dataset view
            # pylint:disable=no-member
            if (
                not len(self._stages)
                and not self._is_clips
                and self._dataset.media_type != fom.GROUP
            ):
                return True

            # Group dataset with all slices selected and no additional stages
            if (
                self._dataset.media_type == fom.GROUP
                and len(self._stages) == 1
                and isinstance(self._stages[0], fos.SelectGroupSlices)
                and not self._pipeline()
            ):
                return True
```